Source/WebCore/platform/graphics/ca/GraphicsLayerCA.cpp

```diff
@@ -743,6 +743,12 @@ void GraphicsLayerCA::setDrawsHDRContent(bool drawsHDRContent)
     GraphicsLayer::setDrawsHDRContent(drawsHDRContent);
     noteLayerPropertyChanged(DrawsHDRContentChanged | DebugIndicatorsChanged);
 }
+
+void GraphicsLayerCA::setNeedsDisplayIfHeadroomExceeds(float headroom)
+{
+    if (protectedLayer()->setNeedsDisplayIfHeadroomExceeds(headroom))
```

Comment:
This is confusing; why does `GraphicsLayerCA` need to call `setNeedsDisplay()` if `setNeedsDisplayIfHeadroomExceeds()` already did it? Maybe `setNeedsDisplayIfHeadroomExceeds()` should just be `needsRepaintIfHeadroomExceeds()`?

Reply:
Forgot to get back to fixing that!

I think it doesn't, and doing so was actively bad, since it'd mean we repaint tiles that don't have HDR content.

We do however need the feedback that something changed, so that we can schedule a flush (rendering update), in case this happened outside of one.